editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/globalconfiguration/beans/BeanConfigUtil.java

```diff
@@ -526,7 +527,8 @@ public void editBeanArgument(Element xmlElement, String type, String value) {
 	}
 	
 	public Element createBeanNode(final CamelFile camelFile, String id, String className) {
-		final String prefixNS = camelFile.getRouteContainer().getXmlNode().getPrefix();
+		// get NS prefix from parent document, not route container node
+		final String prefixNS = camelFile.getRouteContainer().getXmlNode().getOwnerDocument().getPrefix();
```

Comment:
I'm proposing to avoid assumptions letting apart some cases.

I don't see a speed issue with getting the "beans" node prefix as it should be just the parent node to where we want to create it and most the time the top node.